gradle/base/dsl-extensions/build.gradle.kts

```diff
@@ -1,3 +1,14 @@
+import org.jetbrains.kotlin.gradle.dsl.JvmTarget.JVM_21
+
 plugins {
     `kotlin-dsl`
 }
+
+java {
+    sourceCompatibility = JavaVersion.VERSION_21
```

Comment:
please do not use compatibility properties, use release instead.

See https://github.com/apple/pkl/pull/876